src/renderer/pathTracer.h

```diff
@@ -4,29 +4,43 @@
  * license that can be found in the LICENSE file.
  */
 
-#ifndef __RENDERER_H__
-#define __RENDERER_H__
+#ifndef __PATHTRACER_H__
+#define __PATHTRACER_H__
 
+#include <functional>
 #include <glm/glm.hpp>
 #include <math.h>
+#include <memory>
 
 #include "raycast/ray.h"
+#include "renderer/pixelBuffer.h"
 #include "scene/scene.h"
 
-class Renderer {
+class PathTracer {
+  typedef std::function<void()> drawCallBack;
```

Comment:
there is some weird name mangling going on. unify this with src/window/window.hL:15